app/src/main/java/com/example/babytracking/BreastActivity.kt

```diff
@@ -1,11 +1,136 @@
 package com.example.babytracking
 
 import android.os.Bundle
+import android.text.TextUtils
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.TimePicker
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.breast_layout.breast_quality_baby
+import kotlinx.android.synthetic.main.breast_layout.breast_button_save
+import kotlinx.android.synthetic.main.breast_layout.breast_length
+import kotlinx.android.synthetic.main.breast_layout.breast_quality_mother
+import kotlinx.android.synthetic.main.breast_layout.spinner_breast
+
+class BreastActivity : AppCompatActivity() {
+
+    private var hour: Int = 0
+    private var minute: Int = 0
+    private var ampm: String = "AM"
+    private var breast: String = ""
 
-class BreastActivity: AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.breast_layout)
+
+        breastTimeClick()
+        breastSelection()
+        breastSave()
+    }
+
+    private fun breastTimeClick() {
+        val timePicker = findViewById<TimePicker>(R.id.sleep_timepick_hour)
+        timePicker.setOnTimeChangedListener { _, pickerHour, pickerMinute ->
+            hour = pickerHour
+            minute = pickerMinute
+
+            // AM_PM decider logic
+            when {
+                hour == 0 -> {
+                    hour += 12
+                    ampm = "AM"
+                }
+                hour == 12 -> ampm = "PM"
+                hour > 12 -> {
+                    hour -= 12
+                    ampm = "PM"
+                }
+                else -> ampm = "AM"
+            }
```

Comment:
We really shouldn't be storing this as a string. It should be converted to a String just for display, otherwise please store it as a `Long` timestamp.

Here, you can convert it to a timestamp using [`LocalDateTime`](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html) and [`ZonedDateTime`](https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html):

```kt
val localDateTime = LocalDateTime.of(year, month, dayOfMonth, hour, minute) // You'll notice that you need to figure out how to have a user input the date too, as it's possible we'll be inputting an event for a previous day.
val zonedDateTime = ZonedDateTime.of(localDateTime, ZoneId.getSystemDefault())
val timestamp = zonedDateTime.toInstant().toEpochMilli()
```

There are many reasons why this is preferable. One of them being that String require more bits to be stored, so a timestamp will be more efficient. Additionally, the timestamp contains extra data like the date and timezone, and can be converted across timezones if needed. Last, but not least, it's much easier to sort events using a timestamp.